src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -77,8 +77,14 @@ public override IEnumerable<CodeProperty> AddProperty(params CodeProperty[] prop
         if (properties.Length == 0)
             throw new ArgumentOutOfRangeException(nameof(properties));
 
-        return properties.Select(property =>
+        return properties.GroupBy(static x => x.Name).Select(static x => x.First()).Select(property =>
         {
+            // Prevent duplicates
```

Comment:
Those changes are fixing #4178

Please note that two properties with colliding serialization names are going to collide anyway causing just an even more subtle runtime error.
I believe this is an improvement that doesn't impact code that was already correct, but it would be great to test the changes on the Graph API to confirm.

Reply:
To test in other projects I used a pre-release pipeline:
https://github.com/andreaTP/kiota-prerelease/releases/tag/v0.0.0-pre%2BandreaTP.initial-fixes-simple-oneOf-py.b46369a

Reply:
@andreaTP The PR looks good from my end. I just need to run some tests on the graph api metadata and I'll confirm if everything is good here.

Reply:
Without the check, a discriminator property is added multiple times (as reported in other issues) and it breaks serialization/deserialization.

Reply:
and that discriminator property being added multiple times only happens whenever we have a union type?

Reply:
I don't know, it happens for sure when we have a union type, if this is the only case I cannot tell.
IIRC the discriminator gets added +2 times anytime the schema declaring it appears in the OpenAPI definition.

Reply:
You can verify the behavior by using the reproducer: https://github.com/microsoft/kiota/pull/4435/files#diff-2862479147a7917233ba0a4bc0e346b2e297c170b54b431a1c3f511765e2ca53